johnwayodi/sm-server
# #161695595 Cleanup Project, Implement suggested Changes

#### What does this PR do?
Perform the necessary changes suggested, fix validation and arithmetic issues on input.
#### Description of Task to be completed?
1. Ensure Sale Creation is done correctly.
2. Ensure product quantity is reduced after a sale is made containing it.
3. Validate negative input on sale record POST endpoint.
4. Add tests for the above functionality.
#### How should this be manually tested?
1. `git clone https://github.com/johnwayodi/sm-api-v2.git`
2. Ensure you are in the cloned repo, **sm-api-v2** by default.
3. Create a virtual environment using virtualenv and _pip install -r requirements.txt_
4. `git checkout -b ch-cleanup-project-161695595  origin/ch-cleanup-project-161695595`
5. Run the flask application and test using postman.
#### Any background context you want to provide?
The product quantity was not reducing after a sale was made, and the sale creation process didn't check for negative input.
#### What are the relevant pivotal tracker stories?
#161695595
#### Screenshots (if appropriate)
#### Questions:
Any more changes that I can make, drop a comment below
